lib/queue.rb

```diff
@@ -1,31 +1,61 @@
 class Queue
 
-  def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+  attr_accessor :ring, :front, :back, :size
+
+  def initialize(size=20)
+    @size = size
+    @ring = Array.new(size)
+    @front = 0
+    @back = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # if circular buffer is full, return empty array
+    return [] if (@back + 1) % @size == @front
+
+    # otherwise, add element to the back
+    @ring[@back] = element
+
+    # change @back to next free position, clockwise
+    @back = (@back + 1) % @size
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    return [] if @ring.empty?
+
+    dequeued_element = @ring[@front]
+    # dequeue by reassigning the front to the next position, clockwise
+    @front = (@front + 1) % @size
+
+    return dequeued_element
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    @front
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    @size
   end
```

Comment:
Size should be the number of elements in the queue.  You can calculate it using the position of front and rear.